### PR TITLE
Use volume resource for ebs recipe and cleanup of all common libraries

### PR DESCRIPTION
**Pull request by enrico-usai**
### Description of changes
* Use volume resource for ebs recipe
  * I had to move format_directory to environment libraries because explicitly used by the ebs recipe.
* Replace all usage of virtualized with on_docker
* Delete unused graphic_instance function
  * This is defined in platform libraries too and used by nvidia, dcv, slurm recipes.

### Tests
* Moved tests for format_directory method.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-common/test/common/libraries/os_properties.rb [hunk 1]
```diff
@@ -8,13 +8,6 @@ class OsProperties < Inspec.resource(1)
     os_properties.redhat_ubi?
   '
 
-  #
-  # Check if we are running in a virtualized environment
-  #
-  def virtualized?
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-common/test/common/libraries/os_properties.rb:14]: Love it!

File: cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+require 'spec_helper'
+
+describe 'aws-parallelcluster-environment:libraries:format_directory' do
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb:3]: Great test!

File: cookbooks/aws-parallelcluster-environment/recipes/config/ebs.rb [hunk 3]
```diff
@@ -27,24 +27,12 @@
...
-      group 'root'
-      recursive true
-      action :create
-    end
-
-    # Mount shared volume over NFS
-    mount dirname do
-      device(lazy { "#{node['cluster']['head_node_private_ip']}:#{exported_dirname}" })
+    volume "mount volume #{index}" do
+      action :mount
```
Comment (himani2411) [inline cookbooks/aws-parallelcluster-environment/recipes/config/ebs.rb:31]: Wouldnt we need enable action too?

File: cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  cached(:chef_run) do
+    ChefSpec::SoloRunner.new.converge_dsl do
+      volume 'nothing' do
```
Comment (awslitvit) [inline cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb:6]: Not sure this is even required, I guess you can remove the body of the converge.

**Review by awslitvit — APPROVED** [review]

File: cookbooks/aws-parallelcluster-environment/recipes/config/ebs.rb (hunk 3 above)
Comment (enrico-usai) [reply cookbooks/aws-parallelcluster-environment/recipes/config/ebs.rb:31]: We're calling the `mount`action of the `volume` resource that includes both `mount` and `enable` of the `nfs` resource.

File: cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb (hunk 4 above)
Comment (enrico-usai) [reply cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb:6]: Thanks, I'll address it in a separate patch.

File: cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb (hunk 2 above)
Comment (enrico-usai) [reply cookbooks/aws-parallelcluster-environment/spec/unit/libraries/volume_spec.rb:3]: You wrote this! I just moved it :D

**Pull request merged**